### PR TITLE
update triton server version to r24.03

### PR DESCRIPTION
**Pull request by zhyncs**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily receiving feedbacks. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

https://github.com/InternLM/lmdeploy/issues/1363

## Modification

as titled

## Checklist

1. Pre-commit or other linting tools are used to fix the potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit tests to ensure the correctness.
3. If the modification has a dependency on downstream projects of a newer version, this PR should be tested with all supported versions of downstream projects.
4. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

**Comment by irexyc** [comment]
Could you please update the Dockerfile and provide a image on Dockerhub so that we can use it to test the triton server ？

The base image and torch should be updated in Dockerfile

**Comment by zhyncs** [comment]
> Could you please update the Dockerfile and provide a image on Dockerhub so that we can use it to test the triton server ？
> 
> The base image and torch should be updated in Dockerfile

ok I will take a look.

**Comment by zhyncs** [comment]
Hi @irexyc Shall we also update the Python from 3.8 to 3.10 accordingly? And I think https://download.pytorch.org/whl/cu121/torch-2.1.0%2Bcu121-cp310-cp310-linux_x86_64.whl is ok

https://docs.nvidia.com/deeplearning/triton-inference-server/release-notes/rel-24-03.html

> The container also includes the following:
[Ubuntu 22.04](http://releases.ubuntu.com/22.04/) including [Python 3.10](https://www.python.org/downloads/release/python-3100/)
[NVIDIA CUDA 12.4.0.41](https://docs.nvidia.com/cuda/cuda-toolkit-release-notes/index.html)
[NVIDIA cuBLAS 12.4.2.65](https://docs.nvidia.com/cuda/cublas/index.html)
[cuDNN 9.0.0.306](https://docs.nvidia.com/deeplearning/cudnn/release-notes/)
[NVIDIA NCCL 2.20](https://docs.nvidia.com/deeplearning/nccl/release-notes/) (optimized for [NVIDIA NVLink](http://www.nvidia.com/object/nvlink.html)®)

**Comment by zhyncs** [comment]
> Could you please update the Dockerfile and provide a image on Dockerhub so that we can use it to test the triton server ？
> 
> The base image and torch should be updated in Dockerfile

@zhulinJulia24 I don't have an Ubuntu development environment locally, may you help take a look?

https://github.com/InternLM/lmdeploy/pull/1371/commits/2f585ad718f1f3bf137b98206e253f4d30838527

**Comment by zhyncs** [comment]
flash attention whl only has cu122 version https://github.com/Dao-AILab/flash-attention/releases/download/v2.3.6/flash_attn-2.3.6+cu122torch2.1cxx11abiFALSE-cp310-cp310-linux_x86_64.whl

**Comment by zhulinJulia24** [comment]
> > Could you please update the Dockerfile and provide a image on Dockerhub so that we can use it to test the triton server ？
> > The base image and torch should be updated in Dockerfile
> 
> @zhulinJulia24 I don't have an Ubuntu development environment locally, may you help take a look?
> 
> [2f585ad](https://github.com/InternLM/lmdeploy/commit/2f585ad718f1f3bf137b98206e253f4d30838527)

My test machine cuda version is 118, you don't need to change the version

> > Could you please update the Dockerfile and provide a image on Dockerhub so that we can use it to test the triton server ？
> > The base image and torch should be updated in Dockerfile
> 
> @zhulinJulia24 I don't have an Ubuntu development environment locally, may you help take a look?
> 
> [2f585ad](https://github.com/InternLM/lmdeploy/commit/2f585ad718f1f3bf137b98206e253f4d30838527)

I'm looking it. It will take a while.

**Comment by zhyncs** [comment]
```
Processing /root/packages/flash_attn-2.3.6+cu122torch2.1cxx11abiFALSE-cp310-cp310-linux_x86_64.whl
ERROR: Could not install packages due to an OSError: [Errno 2] No such file or directory: '/root/packages/flash_attn-2.3.6+cu122torch2.1cxx11abiFALSE-cp310-cp310-linux_x86_64.whl'
```
Hi @zhulinJulia24 Do you have any ideas? Thanks.

**Comment by zhulinJulia24** [comment]
> ```
> Processing /root/packages/flash_attn-2.3.6+cu122torch2.1cxx11abiFALSE-cp310-cp310-linux_x86_64.whl
> ERROR: Could not install packages due to an OSError: [Errno 2] No such file or directory: '/root/packages/flash_attn-2.3.6+cu122torch2.1cxx11abiFALSE-cp310-cp310-linux_x86_64.whl'
> ```
> 
> Hi @zhulinJulia24 Do you have any ideas? Thanks.

fixed, rerun is triggered.

**Comment by lvhan028** [comment]
Can we make another Dockerfile instead of modifying the current one?

**Comment by zhyncs** [comment]
> Hi, guys.
> 
> I am afraid I cannot merge this PR in v0.3.0 (to be released version)
> 
> The image is used by puyu product now, which is running on cu118 env. 
> 
> 

It's ok. We may merge it in the next release.

**Comment by zhyncs** [comment]
> Can we make another Dockerfile instead of modifying the current one?

I'm going on vacation right now and maybe you can push commits based on my branch. Thanks very much.

**Comment by lvhan028** [comment]
> > Can we make another Dockerfile instead of modifying the current one?
> 
> I'm going on vacation right now and maybe you can push commits based on my branch. Thanks very much.

OK. Have a nice holiday

**Comment by zhyncs** [comment]
Hi guys. I'm back from my vacation, let me see how to make the modifications better.

File: .github/workflows/evaluate.yml
[the remark this answers is not in the document]
Comment (zhyncs) [reply .github/workflows/evaluate.yml:1]: done

File: .github/workflows/daily_ete_test.yml
[the remark this answers is not in the document]
Comment (zhyncs) [reply .github/workflows/daily_ete_test.yml:1]: done

**Comment by zhyncs** [comment]
Hi @zhulinJulia24 Do we need use PyTorch cu121 in `pr_ete_test.yml`?

**Review by zhulinJulia24 — APPROVED** [review]
lgtm

**Comment by zhyncs** [comment]
ping @irexyc

**Comment by lvhan028** [comment]
Here is my suggestion:
1. make another docker file instead of changing the current one, since it is used in our product.
2. define a compile argument to indicate the triton server version, pass it to cmake, not changing it from "r22" to "r24" directly. Default value is "r22"

**Comment by zhyncs** [comment]
> Here is my suggestion:
> 
> 1. make another docker file instead of changing the current one, since it is used in our product.
> 2. define a compile argument to indicate the triton server version, pass it to cmake, not changing it from "r22" to "r24" directly. Default value is "r22"

ok

**Comment by zhyncs** [comment]
> 2. define a compile argument to indicate the triton server version, pass it to cmake, not changing it from "r22" to "r24" directly. Default value is "r22"

Hi @lvhan028 

https://github.com/zhyncs/lmdeploy/blob/26069a21fe56f3a8e04662e78b645ee28cfc96e3/src/turbomind/triton_backend/CMakeLists.txt#L52-L54
```cmake
set(TRITON_BACKEND_REPO_TAG "r24.03" CACHE STRING "Tag for triton-inference-server/backend repo")
set(TRITON_CORE_REPO_TAG "r24.03" CACHE STRING "Tag for triton-inference-server/core repo")
set(TRITON_COMMON_REPO_TAG "r24.03" CACHE STRING "Tag for triton-inference-server/common repo")
```

Currently supported, just need to use `-DTRITON_BACKEND_REPO_TAG=r22.10 -DTRITON_CORE_REPO_TAG=r22.10 -DTRITON_COMMON_REPO_TAG=r22.10` when in use.
And I think r24.03 might be more suitable as the default value.

**Comment by zhyncs** [comment]
Due to no response for a long time, this PR is closed.

**Pull request closed without merging**